cumulus/pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -265,6 +274,19 @@ fn validate_validation_data(
 	);
 }
 
+/// Run the given closure with the externalities and recorder set.
+fn run_with_externalities_and_recorder<B: BlockT, R, F: FnOnce() -> R>(
+	backend: &TrieBackend<B>,
+	recorder: &mut SizeOnlyRecorderProvider<HashingFor<B>>,
+	execute: F,
+) -> R {
+	let mut overlay = sp_state_machine::OverlayedChanges::default();
+	let mut ext = Ext::<B>::new(&mut overlay, backend);
+	recorder.reset();
```

Comment:
Actually realizing that my comment last time was not correct :see_no_evil: 

Given that we reset the `recorder` here, it is fine to also use this function for `check_inherents`. So can you do this? :D It should not be used there, but better than having it fail the validation.